### PR TITLE
Update Instrumentation to metrics v0.28.0

### PR DESCRIPTION
**Pull request by MadVikingGod**
The following Instrumentation packages use the metrics API, and are updated to v0.28.0

- [x] instrumentation/github.com/astaxie/beego/otelbeego
- [x] instrumentation/github.com/astaxie/beego/otelbeego/test
- [x] instrumentation/github.com/gocql/gocql/otelgocql
- [x] instrumentation/github.com/gocql/gocql/otelgocql/test
- [x] instrumentation/host
- [x] instrumentation/net/http/otelhttp
- [x] instrumentation/net/http/otelhttp/test
- [x] instrumentation/runtime

Note:
In v0.27.0 of the metrics API there was otel/metric/metrictest.  This had a MeterProvider that did not use the SDK to record metrics in memory.  That was removed from v0.28.0 and will be replaced in https://github.com/open-telemetry/opentelemetry-go/issues/2722

### CODE REVIEW

File: CHANGELOG.md
Comment (MrAlias) [inline CHANGELOG.md:13]:
```suggestion
- Upgraded packages using otel/metrics v0.27.0 to v0.28.0. (#1977)
```

**Review by MrAlias — APPROVED** [review]

File: instrumentation/runtime/runtime.go
Comment (Aneurysm9) [inline instrumentation/runtime/runtime.go:148]: Should this be three separate callbacks as it was before?  I'm not sure what impact combining them has.

File: instrumentation/github.com/gocql/gocql/otelgocql/test/gocql_test.go
Comment (dashpole) [inline instrumentation/github.com/gocql/gocql/otelgocql/test/gocql_test.go:111]: Is this also being replaced with the in memory exporter from https://github.com/open-telemetry/opentelemetry-go/issues/2722?

**Review by Aneurysm9 — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

File: instrumentation/runtime/runtime.go
Comment (MadVikingGod) [reply instrumentation/runtime/runtime.go:148]: I don't believe that any of the callbacks are run in separate routines.  So this would just dispense with the overhead of tracking 3 different closures and calling each one serially.

File: instrumentation/github.com/gocql/gocql/otelgocql/test/gocql_test.go
Comment (MadVikingGod) [reply instrumentation/github.com/gocql/gocql/otelgocql/test/gocql_test.go:111]: Yes.
```suggestion
	// TODO: Replace with in memory exporter https://github.com/open-telemetry/opentelemetry-go/issues/2722
	// // Check metrics
```

**Comment by MadVikingGod** [comment]
In an effort to enable users to test the new release I will merge this sooner than the 24h window. This already has approvals from the maintainers and 1+ approver.

**Pull request merged**
